### PR TITLE
Validate HTTPS configuration settings

### PR DESCRIPTION
**Pull request by aalbu**
If `http-server.https.enabled` is true, but `http-server.https.keystore.path` is not specified, `HttpServer` blows up with an NPE at startup.  Adding a bit of validation to help diagnose the misconfiguration.

### CODE REVIEW

File: http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java [hunk 1]
```diff
@@ -652,4 +653,10 @@ public HttpServerConfig setHttp2StreamIdleTimeout(Duration http2StreamIdleTimeou
         this.http2StreamIdleTimeout = http2StreamIdleTimeout;
         return this;
     }
+
+    @AssertTrue(message = "PEM keystore path/password must be provided when HTTPS is enabled")
+    public boolean isValidHttpsConfiguration()
```
Comment (findepi) [inline http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java]: `isHttpsConfigurationValid` ?

please move it right after the getter/setters for properties this validates.

File: http-server/src/test/java/io/airlift/http/server/TestHttpServerConfig.java [hunk 2]
```diff
@@ -178,6 +180,58 @@ public void testExplicitPropertyMappings()
...
+                .put("http-server.threads.max-idle-time", "10m")
+                .put("http-server.net.max-idle-time", "20m")
+                .put("http-server.auth.users-file", "/auth")
+                .put("http-server.admin.enabled", "false")
+                .put("http-server.admin.port", "3")
+                .put("http-server.admin.threads.min", "3")
+                .put("http-server.admin.threads.max", "4")
+                .put("http-server.max-request-header-size", "32kB")
+                .put("http-server.max-response-header-size", "57kB")
+                .put("http-server.http2.max-concurrent-streams", "1234")
```
Comment (findepi) [inline http-server/src/test/java/io/airlift/http/server/TestHttpServerConfig.java]: Can this be shorter? Like, only required configuration (few properties)?

File: http-server/src/test/java/io/airlift/http/server/TestHttpServerConfig.java [hunk 3]
```diff
@@ -178,6 +180,58 @@ public void testExplicitPropertyMappings()
         ConfigAssertions.assertFullMapping(properties, expected);
     }
 
+    @Test(expectedExceptions = ConfigurationException.class, expectedExceptionsMessageRegExp = ".*PEM keystore path/password must be provided when HTTPS is enabled.*")
```
Comment (findepi) [inline http-server/src/test/java/io/airlift/http/server/TestHttpServerConfig.java]: It's better to apply try-catch selectively to the particular line you expect to fail.
In Presto we use `org.assertj.core.api.Assertions.assertThatThrownBy` for this.
Please add assertj as a test dep and use here.

File: http-server/src/test/java/io/airlift/http/server/TestHttpServerConfig.java [hunk 4]
```diff
@@ -178,6 +180,58 @@ public void testExplicitPropertyMappings()
...
+    public void testHttpsConfig()
```
Comment (findepi) [inline http-server/src/test/java/io/airlift/http/server/TestHttpServerConfig.java]: `testInvalidHttpsConfiguration`

File: pom.xml [hunk 5]
```diff
@@ -29,6 +29,7 @@
         <dep.packaging.version>${dep.airlift.version}</dep.packaging.version>
         <dep.jetty.version>9.4.14.v20181114</dep.jetty.version>
         <dep.jersey.version>2.26</dep.jersey.version>
+        <dep.assertj-core.version>3.8.0</dep.assertj-core.version>
```
Comment (findepi) [inline pom.xml]: We don't need this. We can use whatever airbase provides (3.5.2 atm)

Comment (aalbu) [reply pom.xml]: Sorry, I thought `assertj` wasn't used in this project.

Comment (findepi) [reply pom.xml]: it wasn't. But airbase provides default dep version, _if_ you use assertj.

**Pull request merged**